server/src/logic/services/recipe.service.ts

```diff
@@ -0,0 +1,52 @@
+import { IDatabase } from "pg-promise";
+import { Recipe, DbError, ERRORS } from "./../../models/index";
+import { ErrorUtils } from "./../../logic/utils/index";
+import { UserService } from "./../../logic/services/index";
+
+export class RecipeService {
+
+  private _GET_BY_ID: string = `select * 
+                                from recipes
+                                where recipes.id = $1`;
+
+  private _SAVE: string = `insert into recipes(
+                             title,
+                             userId,
+                             description,
+                             imageUrl,
+                             created)
+                           values($1, $2, $3, $4, $5)
+                           returning id`;
+
+  constructor (private _db: IDatabase<any>) {}
+
+  getById (id: number): Promise<Recipe> {
+    console.log("CAll getById");
+    let promise = this._db.one(this._GET_BY_ID, [id]);
+    return promise.catch(error => {
+      return ErrorUtils.handleDbNotFound(
+        ERRORS.RECIPE_NOT_FOUND,
```

Comment:
I also had to read a bit about Promises to get this. But this works since each call to then or catch also returns a new Promise. And if you return sth from inside a Promise this gets the value of the newly created Promise. Its really a bit crazy!